ci-templates/.github/workflows/deploy-to-pantheon-live.yml

```diff
@@ -6,50 +6,74 @@ on:
       - production
 
 jobs:
-  build:
+  build-and-sync:
     runs-on: ubuntu-latest
     steps:
-    - name: Perform build steps
-      run: echo "Perform your build steps"
-
-  sync-to-pantheon:
-    needs: build
-    uses: alleyinteractive/.github/.github/workflows/deploy-to-remote-repository.yml@main
-    with:
-      remote_repo: 'ssh://codeserver.dev.PANTHEON-SITE-ID@codeserver.dev.PANTHEON-SITE-ID.drush.in:2222/~/repository.git'
-      remote_branch: 'master' # Notable that this differs from 'production'
-      destination_directory: 'wp-content/'
-      exclude_list: '.git, .gitmodules, node_modules'
-
-    secrets:
-      REMOTE_REPO_SSH_KEY: ${{ secrets.ALLEY_CI_PANTHEON_SSH_KEY }}
-
-  # To autopromote dev -> test -> live
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          cache: 'npm'
+          cache-dependency-path: package-lock.json
+          node-version: 18
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.0
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, gd
+          tools: composer:v2
+          coverage: none
+
+      - name: Install dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: "--no-progress --no-ansi --no-interaction --prefer-dist --no-dev"
+
+      - name: Install npm dependencies
+        run: npm ci
+
+      - name: Run npm build
+        run: npm run build
+
+      - name: Sync to Pantheon
+        uses: alleyinteractive/action-deploy-to-remote-repository@v1
+        with:
+          remote_repo: 'ssh://codeserver.dev.PANTHEON-SITE-ID@codeserver.dev.PANTHEON-SITE-ID.drush.in:2222/~/repository.git'
+          remote_branch: 'master'
+          destination_directory: 'wp-content/'
+          exclude_list: '.git, .github, .gitmodules, node_modules'
+          pantheon: 'true'
+          ssh-key: ${{ secrets.ALLEY_CI_PANTHEON_SSH_KEY }}
+
   deploy-to-pantheon:
     runs-on: ubuntu-latest
-    needs: sync-to-pantheon
+    needs: build-and-sync
+    env:
+      SITE_ID: YOUR-SITE-ID-HERE
```

Comment:
Is this intentional or does an ID need to be passed?

Reply:
SITE_ID seems to be used as a variable expansion in the terminus commands below. Typically we would double-quote the use of a bash variable like "$BASH_VARIABLE" or, in some cases, make it stand out a bit more with some wrapping like ${BASH_VARIABLE}, but all of these should work.

Reply:
It's a bit awkward as someone will need to put the actual id in the `remote_repo` field above, but then use it as a variable below... dunno if there's anything to be done, though

Reply:
Yeah, you're right. I can experiment with adding an ENV var to the above action also. Alternately, we could put the site ID in secrets and access it there? Then we're at least only setting it once.

Reply:
Actually, the remote repo field doesn't contain the same value as `SITE_ID`. The remote repo field contains what looks like a UUID whereas what we are currently storing as `SITE_ID` contains what Pantheon refers to as a "site." I suspect that the terminology is confusing here, so I'll a) hoist these to env vars that are global to the entire workflow and b) update naming convention to be more explicit about what each contains

Reply:
I created a top-level `env` section in each workflow with the configurable bits - `PANTHEON_SITE` which is the human-readable slug, `PANTHEON_SITE_ID` which is the UUID, and `TERMINUS_VERSION` which is currently buried in the URL we're downloading. That makes the actual steps in each file less likely to need to change by users of this repo and it makes the env configuration easier to spot and update.